add the ability to load from file a DER based certificate

marco-introini opened the pull request:
With this little improvement it's now possible to load a file in which the certificate used is in binary DER based format instead of classic PEM format.

freekmurze commented on src/helpers.php:
```diff
@@ -44,3 +44,9 @@ function str_contains(string $haystack, string | array $needles): bool
     return false;
 }
+
+function der2pem($der_data): string
```
Let's not pollute the global namespace with this function, but instead add it to the `SslCertificate` class

freekmurze commented on tests/SslCertificateFromFileTest.php:
```diff
@@ -0,0 +1,43 @@
+use Spatie\Snapshots\MatchesSnapshots;
+use Spatie\SslCertificate\SslCertificate;
+
+class SslCertificateFromFileTest extends TestCase
```
I appreciate these tests 👍

marco-introini commented:
I moved the der2pem function inside the SslCertificate class as a public static function.
Previous test didn't work for connection error on site with different port. I found another one (casually) which works for now and modified them.
Hope this can help

Thank you so much

freekmurze commented:
Thanks!
